common/djangoapps/third_party_auth/tests/test_saml.py

```diff
@@ -32,3 +35,13 @@ def test_get_user_details(self):
         """ test get_attr and get_user_details of EdXSAMLIdentityProvider"""
         edx_saml_identity_provider = EdXSAMLIdentityProvider('demo', **mock_conf)
         assert edx_saml_identity_provider.get_user_details(mock_attributes) == expected_user_details
+
+
+class TestSAMLAuthBackend(SAMLTestCase):
+
+    @mock.patch('common.djangoapps.third_party_auth.saml.SAMLAuth.auth_complete')
+    def test_saml_auth_complete(self, super_auth_complete):
+        super_auth_complete.side_effect = MultiValueDictKeyError('RelayState')
+        backend = SAMLAuthBackend()
+        with self.assertRaises(AuthMissingParameter):
```

Comment:
Could you grab the exception and check that it has "RelayState" as the missing parameter?